crates/context_aware_config/src/api/context/helpers.rs

```diff
@@ -57,6 +57,40 @@ pub fn validate_condition_with_mandatory_dimensions(
     Ok(())
 }
 
+pub fn validate_condition_with_dependent_dimensions(
+    conn: &mut DBConnection,
+    context: &Condition,
+    schema_name: &SchemaName,
+) -> superposition::Result<()> {
+    use dimensions::dsl;
+    let context_map = extract_dimensions(context)?;
+    for (key, _) in context_map.iter() {
+        let dependency_list: Option<Value> = dsl::dimensions
+            .filter(dsl::dimension.eq(key))
+            .select(dsl::dependency_graph)
+            .schema_name(schema_name)
+            .get_result::<Option<Value>>(conn)?;
+
+        let dependency_list: Vec<String> = dependency_list
+            .iter()
+            .filter_map(|val| val.as_object())
+            .flat_map(|val| val.keys().cloned())
+            .filter(|val| val != key)
+            .collect();
+        let all_dependencies_present = dependency_list
+            .iter()
+            .all(|dimension| context_map.contains_key(dimension));
+        if !all_dependencies_present {
+            return Err(validation_error!(
+                "The context should contain all the dependent dimensions for {} : {:?}.",
+                key,
+                dependency_list,
+            ));
+        }
+    }
```

Comment:
I think you can avoid multiple queries to the db here
```suggestion
    let keys = context_map.keys();
    let dependency_lists: Option<Value> = dsl::dimensions
            .filter(dsl::dimension.eq_any(keys))
            .select(dsl::dependency_graph)
            .schema_name(schema_name)
            .get_result::<Option<Value>>(conn)?;
            
            // rest of code
```